monticore-generator/src/main/java/de/monticore/codegen/parser/antlr/AntlrTool.java

```diff
@@ -119,19 +119,18 @@ public void processNonCombinedGrammar(Grammar g, boolean gencode) {
    */
   private void calculateStatesForNonTerminals(Grammar g) {
     if(g.isParser() || g.isCombined()){
-      for (Map.Entry<ASTProd, Map<ASTNode, String>> outer : tmpNameDict.entrySet()) {
-        if(!outer.getValue().isEmpty()) {
-          for (Map.Entry<ASTNode, String> inner : outer.getValue().entrySet()) {
+      for (Map.Entry<ASTProd, ProdInfo> outer : prodInfo.entrySet()) {
+        Map<ASTNode, String> names = outer.getValue().tmpNames;
+
+        if(!names.isEmpty()) {
+          for (Map.Entry<ASTNode, String> inner : names.entrySet()) {
             ASTNode key = inner.getKey();
-            if(key instanceof ASTNonTerminal){
-              ASTNonTerminal nonTerminal = (ASTNonTerminal) key;
-              String ruleName = outer.getKey().getName();
-              ruleName = ruleName.substring(0, 1).toLowerCase() + ruleName.substring(1);
-              nonTerminalToParserStates.put(
-                      nonTerminal,
-                      calculateStateForTmpName(g, ruleName, inner.getValue())
-              );
-            }
+            String ruleName = outer.getKey().getName();
+            ruleName = ruleName.substring(0, 1).toLowerCase() + ruleName.substring(1);
```

Comment:
```suggestion
            ruleName = StringTransformations.uncapitalize(ruleName);
```
damit es einheitlich ist (ggf muss de.se_rwth.commons.StringTransformations importiert werden)

Reply:
✔